src/components/Favorites/Favorites.test.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { MemoryRouter as Router } from "react-router-dom";
+import { cleanup, render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import Favorites from "./Favorites";
+
+describe("Favorites", () => {
+  afterEach(cleanup);
+
+  it("should render message in Favorites if user has no favorite listings", () => {
+    const mockFavorites = [];
+    const { getByText } = render(
+      <Router>
+        <Favorites favorites={mockFavorites} />
+      </Router>
+    );
+    expect(getByText("My Favorites")).toBeInTheDocument();
+    expect(
+      getByText("You have no favorites yet. Favorite a listing!")
+    ).toBeInTheDocument();
+  });
+
```

Comment:
Maybe test the button functionality for Return to Listings as well?

Reply:
Would this be an integration test in App because we are going back to another component?